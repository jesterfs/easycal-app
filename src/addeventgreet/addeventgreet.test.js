import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import AddEventGreet from './addeventgreet'
import renderer from 'react-test-renderer';

describe('AddEventGreet component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <AddEventGreet />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders the UI as expected', () => {
    const div = document.createElement('div')
    const tree = renderer
      .create(<BrowserRouter>
        <AddEventGreet />
      </BrowserRouter>,
      div)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

});