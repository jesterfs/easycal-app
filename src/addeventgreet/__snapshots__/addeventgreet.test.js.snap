// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEventGreet component renders the UI as expected 1`] = `
<div
  className="AddEventGreet, greetgroup"
>
  <div
    className="item"
  >
    <h2>
      Add an Event
    </h2>
  </div>
  <div
    className="item"
  >
    <form
      className="add-event-form"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="eventName"
        >
          Event Name
        </label>
        <input
          id="eventName"
          name="eventName"
          placeholder="Event Name"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="eventDate"
        >
          Date
        </label>
        <input
          id="eventDate"
          name="eventDate"
          type="date"
        />
      </div>
      <div>
        <label
          htmlFor="eventStartTime"
        >
          Start Time
        </label>
        <input
          id="eventStartTime"
          name="eventStartTime"
          type="time"
        />
        <label
          htmlFor="eventEndTime"
        >
          End Time
        </label>
        <input
          id="eventEndTime"
          name="eventEndTime"
          type="time"
        />
      </div>
      <div>
        <label
          htmlFor="eventParticipants"
        >
          Event Participants:
        </label>
        <br />
        <select
          id="eventParticipants"
          multiple={true}
          name="eventParticipants"
        />
      </div>
      <button
        className="addbtn"
        type="submit"
      >
        Schedule Event
      </button>
    </form>
  </div>
</div>
`;
